contracts/Manager.sol

```diff
@@ -669,6 +692,79 @@ contract Manager is UUPSOwnableUpgradeable, UsingRegistryUpgradeable {
         }
     }
 
+    /**
+     * @notice Votes on a proposal in the referendum stage.
+     * @param proposalId The ID of the proposal to vote on.
+     * @param index The index of the proposal ID in `dequeued`.
+     * @param yesVotes The yes votes weight.
+     * @param noVotes The no votes weight.
+     * @param abstainVotes The abstain votes weight.
+     */
+    function voteProposal(
+        uint256 proposalId,
+        uint256 index,
+        uint256 yesVotes,
+        uint256 noVotes,
+        uint256 abstainVotes
+    ) public {
+        IVote vote = IVote(voteContract);
+
+        (
+            uint256 stCeloUsedForVoting,
+            uint256 totalYesVotes,
+            uint256 totalNoVotes,
+            uint256 totalAbstainVotes
+        ) = vote.voteProposal(msg.sender, proposalId, yesVotes, noVotes, abstainVotes);
+
+        stakedCelo.lockVoteBalance(msg.sender, stCeloUsedForVoting);
+        account.voteProposal(proposalId, index, totalYesVotes, totalNoVotes, totalAbstainVotes);
```

Comment:
Maybe I'm missing it, but can't find the check the proposal we're voting for is in referendum.

Reply:
This is being checked directly in governance smart contract